homeassistant/components/linky/sensor.py

```diff
@@ -49,45 +51,53 @@
     }
 )
 
+_LOGGER.error("LINKY_SENSOR")
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Old way of setting up the Linky platform."""
+    pass
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Configure the platform and add the Linky sensor."""
-    username = config[CONF_USERNAME]
-    password = config[CONF_PASSWORD]
-    timeout = config[CONF_TIMEOUT]
 
-    account = LinkyAccount(hass, add_entities, username, password, timeout)
-    add_entities(account.sensors, True)
+async def async_setup_entry(
+    hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Add Linky entries."""
+    _LOGGER.error("LINKY_SENSOR:async_setup_entry")
+    account = LinkyAccount(
+        entry.data[CONF_USERNAME], entry.data[CONF_PASSWORD], entry.data[CONF_TIMEOUT]
+    )
+
+    await hass.async_add_executor_job(account.update_linky_data())
```

Comment:
Don't call the `update_linky_data` method, just pass it as a target to `hass.async_add_executor_job`.

Reply:
Yep ... my bad

(And I couldn't test since late dev hours and Enedis API not working)